crates/subspace-service/src/domains.rs

```diff
@@ -1,229 +1,33 @@
-use parity_scale_codec::{Decode, Encode};
-use sc_network::{IfDisconnected, NetworkRequest, PeerId, RequestFailure};
-use sc_network_sync::SyncingService;
-use sp_blockchain::HeaderBackend;
-use sp_domains::{DomainId, ExecutionReceiptFor};
-use sp_runtime::traits::{Block as BlockT, Header};
-use std::sync::Arc;
-use std::time::Duration;
-use tokio::time::sleep;
-use tracing::{debug, error, trace};
-
-pub(crate) mod request_handler;
-
-const REQUEST_PAUSE: Duration = Duration::from_secs(5);
-
-/// Last confirmed domain block info error
-#[derive(Debug, thiserror::Error)]
-pub enum LastConfirmedDomainBlockResponseError {
-    #[error("Last confirmed domain block info request failed: {0}")]
-    RequestFailed(#[from] RequestFailure),
-
-    #[error("Last confirmed domain block info request canceled")]
-    RequestCanceled,
-
-    #[error("Last confirmed domain block info request failed: invalid protocol")]
-    InvalidProtocol,
-
-    #[error("Failed to decode response: {0}")]
-    DecodeFailed(String),
-}
-
-#[async_trait]
-pub trait LastDomainBlockReceiptProvider<Block: BlockT, CBlock: BlockT>: Send {
-    async fn get_execution_receipt(
-        &self,
-        block_hash: Option<CBlock::Hash>,
-    ) -> Option<ExecutionReceiptFor<Block::Header, CBlock, Balance>>;
-}
-
-#[async_trait]
-impl<Block: BlockT, CBlock: BlockT> LastDomainBlockReceiptProvider<Block, CBlock> for () {
-    async fn get_execution_receipt(
-        &self,
-        _: Option<CBlock::Hash>,
-    ) -> Option<ExecutionReceiptFor<Block::Header, CBlock, Balance>> {
-        None
-    }
-}
+#![warn(missing_docs)]
 
-#[async_trait]
-impl<Block, CBlock, Client, NR> LastDomainBlockReceiptProvider<Block, CBlock>
-    for LastDomainBlockInfoReceiver<Block, Client, NR>
-where
-    Block: BlockT,
-    CBlock: BlockT,
-    NR: NetworkRequest + Sync + Send,
-    Client: HeaderBackend<Block>,
-{
-    async fn get_execution_receipt(
-        &self,
-        block_hash: Option<CBlock::Hash>,
-    ) -> Option<ExecutionReceiptFor<Block::Header, CBlock, Balance>> {
-        self.get_last_confirmed_domain_block_receipt::<CBlock>(block_hash)
-            .await
-    }
-}
+pub mod snap_sync_orchestrator;
 
-pub struct LastDomainBlockInfoReceiver<Block, Client, NR>
-where
-    Block: BlockT,
-    NR: NetworkRequest,
-    Client: HeaderBackend<Block>,
-{
-    domain_id: DomainId,
-    fork_id: Option<String>,
-    client: Arc<Client>,
-    network_service: NR,
-    sync_service: Arc<SyncingService<Block>>,
-}
+use crate::domains::snap_sync_orchestrator::SnapSyncOrchestrator;
+use crate::FullBackend;
+use sc_consensus_subspace::SubspaceLink;
+use sc_network::NetworkRequest;
+use sc_network_sync::SyncingService;
+use sp_runtime::traits::Block as BlockT;
+use std::sync::Arc;
 
-impl<Block, Client, NR> LastDomainBlockInfoReceiver<Block, Client, NR>
+/// Provides parameters for domain snap sync synchronization with the consensus chain snap sync.
+pub struct ConsensusChainSyncParams<Block, CNR>
```

Comment:
Everything included in this struct is already public and can be accessed in `subspace-node`, this struct should not exist in `subspace-service`, it doesn't belong here.

Reply:
I answered above: the question remains about the dependency for block import notification stream.

Reply:
Stream is just a stream. Domains already use some of the streams that are derived from consensus logic, yet they don't depend (I hope) on consensus crate for this.